This pull request adds support for several RegEx character classes

corsonknowles opened the pull request:
Newly included character classes: 
    // "\d" any non-digit character
    // \b	word boundary
    // \B not-word boundary
    // carriage return \r

The above comments can be removed from methodMatch.js, and are only included to make code review easier. 

Newly added:
  support for negation of a group: none of abc -> [^abc]

Removed these RegEx characters that are not supported in JavaScript
    // \A start of string, contrast with ^ which breaks at new lines
    // \z end of string, contrast with $ which breaks at new lines

corsonknowles committed:
Comment in character classes to be added

corsonknowles committed:
Add carriage return placeholder comment

corsonknowles committed:
Add helper methods and builder hashes for additonal RegEx character classes

corsonknowles committed:
Add all new character classes, clean up comments

KarimGeiger commented:
Hi @corsonknowles, thanks again for this contribution. It looks good to me. 

@teabyii, what's your opinion here? Are you missing something, or can we merge those changes?

corsonknowles committed:
Remove A and z character classes which are not supperted in JavaScript, add tests for new characters and improved helper methods

corsonknowles committed:
Modify test rules

teabyii commented on package.json:
```diff
@@ -33,5 +33,8 @@
     "mocha-lcov-reporter": "^1.2.0"
+  },
+  "dependencies": {
+    "test": "^0.6.0"
```
I don't think it needs this dependency.

teabyii reviewed (commented):
Remove `package-lock.json`？

teabyii commented:
Thank you for the PR, and try to increase the code coverage?

corsonknowles committed:
Gitignore package lock and remove test package

corsonknowles committed:
Remove nested git file

corsonknowles committed:
Move rules back into place

corsonknowles committed:
Remove git submodule

corsonknowles committed:
change file encoding of new test rules

corsonknowles committed:
Amend rules files for word and no word

corsonknowles committed:
Change allowed type of word boundaries

corsonknowles committed:
Add mapper functions for word and nonword boundaries

corsonknowles committed:
Fix all new test rules, verify working

corsonknowles committed:
Hard delete package-lock

teabyii commented:
Why remove submodule and add `Test-Rules` code directly to this repo? I don't think it's a good way to maintenance the common test rules for different languages.

corsonknowles commented:
Ah. I think I misunderstood the requests to add tests and increase coverage
and also the comment about removing the dependency. What is the correct
approach? Karim also asked me to write rules for the new RegEx characters
supported, and I followed the readme docs for the .rule files

I would be very happy to finish this today. Ideally, we will be giving a
demonstration with SRL-JavaScript tomorrow, and it would be very nice to
get to demonstrate it with these few minor updates in place.




On Sat, Aug 12, 2017 at 8:13 AM Boom Lee <notifications@github.com> wrote:

> Why remove submodule and add Test-Rules code directly to this repo? I
> don't think it's a good way to maintenance the common text rules for
> different languages.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/SimpleRegex/SRL-JavaScript/pull/9#issuecomment-321987302>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABpRy1EifYXna5y1rdDgOfkM2MMPeeJqks5sXcElgaJpZM4Oygj5>
> .
>

corsonknowles committed:
Restore .gitmodules for test

corsonknowles committed:
Update package.json

corsonknowles committed:
Move rules files to submodule repo

corsonknowles committed:
remove duplicate rules files

corsonknowles committed:
Remove only rules file duplicates

corsonknowles committed:
Reactivate submodule

corsonknowles commented:
Ah, I see. I have moved the .rules files to a new Pull Request for the
relevant submodule in the rules repo.

I removed the local copy of the .rules files, and the changes to the
submodule for SRL-JavaScript.

Thanks!

On Sat, Aug 12, 2017 at 10:17 AM, David Corson-Knowles <
david.corsonknowles@gmail.com> wrote:

> Ah. I think I misunderstood the requests to add tests and increase
> coverage and also the comment about removing the dependency. What is the
> correct approach? Karim also asked me to write rules for the new RegEx
> characters supported, and I followed the readme docs for the .rule files
>
> I would be very happy to finish this today. Ideally, we will be giving a
> demonstration with SRL-JavaScript tomorrow, and it would be very nice to
> get to demonstrate it with these few minor updates in place.
>
>
>
>
> On Sat, Aug 12, 2017 at 8:13 AM Boom Lee <notifications@github.com> wrote:
>
>> Why remove submodule and add Test-Rules code directly to this repo? I
>> don't think it's a good way to maintenance the common text rules for
>> different languages.
>>
>> —
>> You are receiving this because you were mentioned.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/SimpleRegex/SRL-JavaScript/pull/9#issuecomment-321987302>,
>> or mute the thread
>> <https://github.com/notifications/unsubscribe-auth/ABpRy1EifYXna5y1rdDgOfkM2MMPeeJqks5sXcElgaJpZM4Oygj5>
>> .
>>
>

corsonknowles committed:
Add vertical tab as supported JavaScript regex character, and remove my comments from methodMatch

corsonknowles committed:
Add support for text input of backslash

KarimGeiger commented on lib/Builder.js:
```diff
@@ -39,16 +39,31 @@ const simpleMapper = {
         'type': METHOD_TYPE_CHARACTER,
         'allowed': METHOD_TYPES_ALLOWED_FOR_CHARACTERS
     },
+    'vertical tab': {
```
Please stick with the camelCase convention there.

corsonknowles committed:
Amend vertical tab in mbuilder and methodMatch

corsonknowles commented:
Thanks! Corrected.

On Sun, Aug 13, 2017 at 6:17 AM, David Corson-Knowles / CloudFunded <
david@cloudfunded.com> wrote:

> Thanks! Corrected.
>
> On Sun, Aug 13, 2017 at 6:03 AM, Karim Geiger <notifications@github.com>
> wrote:
>
>> *@KarimGeiger* commented on this pull request.
>> ------------------------------
>>
>> In lib/Builder.js
>> <https://github.com/SimpleRegex/SRL-JavaScript/pull/9#discussion_r132841710>
>> :
>>
>> >      'tab': {
>>          'add': '\\t',
>>          'type': METHOD_TYPE_CHARACTER,
>>          'allowed': METHOD_TYPES_ALLOWED_FOR_CHARACTERS
>>      },
>> +    'vertical tab': {
>>
>> Please stick with the camelCase convention there.
>>
>> —
>> You are receiving this because you were mentioned.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/SimpleRegex/SRL-JavaScript/pull/9#pullrequestreview-55965591>,
>> or mute the thread
>> <https://github.com/notifications/unsubscribe-auth/ABpRy8u5wQPh9O3uPUMqp-KrkQoFz9w-ks5sXvQrgaJpZM4Oygj5>
>> .
>>
>
>

teabyii commented:
Well done. Now Wait for [PR for Test-rules](https://github.com/SimpleRegex/Test-Rules/pull/2) to be merged.

corsonknowles commented:
Thank you! Rules just merged.

On Sun, Aug 13, 2017 at 11:20 PM Boom Lee <notifications@github.com> wrote:

> Well done. Now Wait for PR for Test-rules
> <https://github.com/SimpleRegex/Test-Rules/pull/2> to be merged.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/SimpleRegex/SRL-JavaScript/pull/9#issuecomment-322108695>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABpRy_pGxMkuf93fzLIDUrPyrlnAGIEhks5sX-cogaJpZM4Oygj5>
> .
>

teabyii commented:
@corsonknowles Reorganize the commits and put them in one commit please? That looks more beautiful, thank you.

KarimGeiger commented:
@teabyii you can do this using GitHub while merging. Just select "Squash and merge" to combine them.

teabyii commented:
Ok, I got it.
